url.py

```diff
@@ -322,6 +322,10 @@ def unpunycode(self):
     ###########################################################################
     # Information about the domain
     ###########################################################################
+    def hostname(self):
```

Comment:
Maybe you just want to alias `punycode` instead. I don't think you want the return value to be either punycoded or unpunycoded based on cnotext.